### PR TITLE
Use S3 for sending large messages

### PR DESCRIPTION
**Pull request by chendoron**

### CODE REVIEW

File: lib/records.js [hunk 1]
```diff
@@ -87,18 +100,53 @@ function getRecordsDecoder(compression, inputEncoding) {
...
+
+    if (useS3ForLargeItems && data.length > largeItemThreshold * 1024) {
+      const s3ItemKey = `${shortUuid.generate()}.json`;
+      const putData = await s3Client.putObject({
+        Body: data,
+        Bucket: bucketName,
+        Key: s3ItemKey
+      });
+
+      console.log(`need to handle nonS3Keys here`, nonS3Keys);
```
Comment (chendoron) [inline lib/records.js]: Finish up nonS3Keys logic

File: lib/index.js [hunk 2]
```diff
@@ -165,6 +171,17 @@ class Kinesis extends PassThrough {
    * @param {number} [options.pollDelay=250] - When the `usePausedPolling` option is `false`, this
    *        option defines the delay in milliseconds in between poll requests for more records
    *        (only applicable when `useEnhancedFanOut` is set to `false`)
+   * @param {object} [options.s3={}] - The initialization options for the S3 client used
+   *        to store large items in buckets. In addition to `bucketName` and `endpoint`, it
+   *        can also contain any of the [`AWS.S3` options]{@link external:AwsJsSdkS3}.
+   * @param {string} [options.s3.bucketName=options.streamName] - The name of the bucket in which to store
```
Comment (chendoron) [inline lib/index.js]: Remove default doc and explain

File: package.json [hunk 3]
```diff
@@ -58,11 +58,13 @@
     "lru-cache": "^5.1.1",
     "lzutf8": "^0.5.5",
     "project-name": "^1.0.0",
-    "short-uuid": "^3.1.1"
+    "short-uuid": "^3.1.1",
+    "uuid": "^3.3.3"
```
Comment (eaviles) [inline package.json]: Please remove "uuid" as a dependency

File: package.json [hunk 4]
```diff
@@ -58,11 +58,13 @@
...
   },
   "devDependencies": {
     "auto-changelog": "^1.16.0",
     "chalk": "^2.4.2",
+    "chance": "^1.0.18",
```
Comment (eaviles) [inline package.json]: I don't see it in use. Remove?

**Pull request merged**